build/deploy/cockroach.sh

```diff
@@ -257,64 +251,28 @@ create_default_user() {
   fi
 }
 
-# check_if_server_fully_stopped is to wait until the init server is fully
-# stopped or timeout.
-check_if_server_fully_stopped() {
-  local log_path=./cockroach-data/logs/cockroach.log
-  local timeout=20
-  local time_counter=0
-
-  until [[ ( $time_counter -eq $timeout ) || ( -f $log_path ) ]]; do
-    echo >&2 "$log_path doesn't exist, waiting ..."
-    sleep 2
-    time_counter=$((time_counter+1))
-  done
-
-  # If timeout, exit the program.
-  if [[ $time_counter -ge $timeout ]]; then
-    echo >&2 "error: timeout for finding log file"
-    exit 1
-  fi
-
-  time_counter=0
-  echo >&2 "waiting for the init server to be fully stopped..."
-
-  # Wait until either the last line of the log contains "server drained and
-  # shutdown completed" or timeout.
-  until [[ ( $time_counter -eq $timeout ) || \
-        ( "$(tail -1 $log_path)" == *"server drained and shutdown completed"* ) ]];
-  do
-    sleep 2
-    time_counter=$((time_counter+1))
-  done
-
-  # If timeout, exit the program.
-  if [[ $time_counter -ge $timeout ]]; then
-    echo >&2 "error: timeout for stopping the init server"
-    exit 1
-  fi
-  echo >&2 "init server fully stopped"
-}
-
 # run_single_node process the command if it contains `start-single-node` argument.
 run_single_node() {
   # If /cockroach-data is empty, run the initialization steps.
   if [[ $(ls -A cockroach-data | wc -l) = 0 ]]; then
     setup_certs_dir
     setup_env
-    # Start the init server.
+    # Start the server.
     start_init_node "$@"
     setup_db "$@"
     process_init_files /docker-entrypoint-initdb.d/*
-    # Stop the init server.
-    stop_init_node
-    touch init_success
+    # Bring the background server process to the foreground, otherwise the
+    # docker container will automatically exit here.
+    echo "init_finished" > ./cockroach-data/init_success
+    fg %1
+  else
+      # Start the real server.
```

Comment:
"real server" is probably not a necessary comment -- since the `if` branch above is also a "real" server now

Reply:
Done.